src/jinja2/bccache.py

```diff
@@ -262,13 +264,73 @@ def _get_cache_filename(self, bucket: Bucket) -> str:
     def load_bytecode(self, bucket: Bucket) -> None:
         filename = self._get_cache_filename(bucket)
 
-        if os.path.exists(filename):
-            with open(filename, "rb") as f:
-                bucket.load_bytecode(f)
+        # Don't test for existence before opening the file, since
+        # doing so is racy: The file could disappear after the test
+        # and before the open.  Since we don't want to catch
+        # FileNotFound in the event that load_bytecode raises it, we
+        # do not use 'with'.
+        try:
+            f = open(filename, "rb")
+        except FileNotFoundError:
+            return
+        try:
+            bucket.load_bytecode(f)
+        finally:
+            f.close()
```

Comment:
This could use `with f: ...` instead of the try..finally, couldn't it?

Reply:
If we use `with f`, it will also mean catching any FileNotFound errors raised by load_bytecode, instead of just the open.  It's possible a subclass could change load_bytecode to do more complex things that include opening files.

Reply:
I don't think `with f` catches any errors, does it? I mean keeping the open outside:

```py
f = open(...)
with f:
    bucket.load_bytecode(f)
```

Reply:
nevermind, I understand what you are suggesting.